test/membrane_rtmp_plugin/rtmp_sink_test.exs

```diff
@@ -45,26 +50,50 @@ defmodule Membrane.RTMP.SinkTest do
   test "Check if audio and video streams are correctly received by RTMP server instance", %{
     flv_output_file: flv_output_file
   } do
-    rtmp_server = Task.async(fn -> start_rtmp_server(flv_output_file) end)
+    test_stream_correctly_received(flv_output_file, @reference_flv_path, [:audio, :video])
+  end
 
-    sink_pipeline_pid = start_sink_pipeline(@rtmp_server_url)
+  @tag :tmp_dir
+  test "Check if a single audio track is correctly received by RTMP server instance", %{
+    flv_output_file: flv_output_file
+  } do
+    test_stream_correctly_received(flv_output_file, @reference_flv_audio_path, [:audio])
+  end
+
+  @tag :tmp_dir
+  test "Check if a single video track is correctly received by RTMP server instance", %{
+    flv_output_file: flv_output_file
+  } do
+    test_stream_correctly_received(flv_output_file, @reference_flv_video_path, [:video])
+  end
+
+  defp test_stream_correctly_received(flv_output, fixture, tracks) do
+    rtmp_server = Task.async(fn -> start_rtmp_server(flv_output) end)
+
+    sink_pipeline_pid = start_sink_pipeline(@rtmp_server_url, tracks)
 
     # There's an RC - there's no way to ensure RTMP server starts to listen before pipeline is started
     # so it may retry a few times before succeeding
     assert_pipeline_play(sink_pipeline_pid, 5000)
 
-    assert_start_of_stream(sink_pipeline_pid, :rtmp_sink, :video, 5_000)
-    assert_start_of_stream(sink_pipeline_pid, :rtmp_sink, :audio, 5_000)
+    if :audio in tracks,
+      do: assert_start_of_stream(sink_pipeline_pid, :rtmp_sink, Pad.ref(:audio, 0), 5_000)
+
+    if :video in tracks,
+      do: assert_start_of_stream(sink_pipeline_pid, :rtmp_sink, Pad.ref(:video, 0), 5_000)
+
+    if :audio in tracks,
+      do: assert_end_of_stream(sink_pipeline_pid, :rtmp_sink, Pad.ref(:audio, 0), 5_000)
 
-    assert_end_of_stream(sink_pipeline_pid, :rtmp_sink, :video, 5_000)
-    assert_end_of_stream(sink_pipeline_pid, :rtmp_sink, :audio, 5_000)
+    if :video in tracks,
+      do: assert_end_of_stream(sink_pipeline_pid, :rtmp_sink, Pad.ref(:video, 0), 5_000)
 
     :ok = Pipeline.terminate(sink_pipeline_pid, blocking?: true)
     # RTMP server should terminate when the connection is closed
     assert :ok = Task.await(rtmp_server)
 
-    assert File.exists?(flv_output_file)
-    assert File.stat!(flv_output_file).size == File.stat!(@reference_flv_path).size
+    assert File.exists?(flv_output)
+    assert File.stat!(flv_output).size == File.stat!(fixture).size
```

Comment:
Hmm, for one it doesn't work. Here's the diff between the fixture and the output for a single video track test (btw, that's the only test which fails): 
<img width="1812" alt="image" src="https://github.com/membraneframework/membrane_rtmp_plugin/assets/48837433/b0b05ebb-182e-4263-9f09-698c748a2fba">

Reply:
Since the single video track test is newly added test, perhaps it would be ok to make a "snapshot" test and provide the output of our pipeline as the reference result? Surely, we should also manually confirm the the output .flv of our pipeline is correct. Now it won't check anything, but it will bring benefits with further changes so that to make sure the result is not changing.